src/coreclr/src/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -507,7 +572,7 @@ private void MergeWithGlobalList(T[] list)
                             }
```

Comment:
Nit, this method could maybe use `Debug.Assert(Monitor.IsEntered(this))` at the top

Reply:
Honestly I wanted to make this a normal `lock`. :)

Reply:
Such asserts do seem to find bugs from time to time, there was one we fixed a couple months ago. So I guess there can sometimes be value in using Monitor to make this possible when a lock protects callees.

Reply:
One could imagine an analyzer validating it one day?